src/poprox_storage/repositories/articles.py

```diff
@@ -229,6 +230,62 @@ def fetch_entity_by_name(self, entity_name: str) -> Entity | None:
             raw_data=row.raw_data,
         )
 
+    def fetch_latest_package_by_entity(self, entity_id: UUID) -> ArticlePackage:
+        packages_table = self.tables["article_packages"]
+        contents_table = self.tables["article_package_contents"]
+        entities_table = self.tables["entities"]
+
+        query = (
+            select(packages_table)
+            .where(packages_table.c.entity_id == entity_id)
+            .order_by(desc(packages_table.c.created_at))
+            .limit(1)
+        )
+
+        package_rows = self.conn.execute(query).fetchall()
```

Comment:
Since we expect the query above to return at most one row, I'd use `one_or_none()` instead of `fetchall()` here. The advantages are that:
* it will fail loudly with an error if the query above somehow returns more than one row
* the code below won't have to iterate through rows (because there's at most one)
```suggestion
        package_row = self.conn.execute(query).one_or_none()
```